### ISSUE TITLE
Spell checking fails on Android

### ISSUE THREAD

**Issue by Voklen** [flutter/flutter#117167]
## Steps to Reproduce

I have made a [diary app](https://github.com/Voklen/Daily-Diary  
) with spell checking that used to work perfectly. But now it no longer does and gives an error, even when going to the commit it worked and setting Flutter back to 3.6.0-1.0.pre it still errors.

1. Execute `flutter run` on the code sample
2. Click on text field
3. Text is not corrected and outputs:

<details>
<summary>Output</summary>

```
[+2696 ms] E/MethodChannel#flutter/spellcheck(21756): Failed to handle method call
[        ] E/MethodChannel#flutter/spellcheck(21756): java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.textservice.SpellCheckerSession.getSentenceSuggestions(android.view.textservice.TextInfo[], int)' on a null object reference
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.plugin.editing.SpellCheckPlugin.performSpellCheck(SpellCheckPlugin.java:105)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.plugin.editing.SpellCheckPlugin.initiateSpellCheck(SpellCheckPlugin.java:87)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.embedding.engine.systemchannels.SpellCheckChannel$1.onMethodCall(SpellCheckChannel.java:63)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:258)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at android.os.Handler.handleCallback(Handler.java:938)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at android.os.Handler.dispatchMessage(Handler.java:99)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at android.os.Looper.loop(Looper.java:223)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at android.app.ActivityThread.main(ActivityThread.java:7715)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at java.lang.reflect.Method.invoke(Native Method)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
[        ] E/MethodChannel#flutter/spellcheck(21756): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
```
</details>

**Expected results:**
Red underlines

**Actual results:** 
No change whether the word is spelled correctly or not
<img src="https://user-images.githubusercontent.com/56766748/207962140-4002bd3d-5204-4536-b61f-cad2b262fd8e.png" alt="Screenshot of random text that has not been highlighted as being incorrectly spelled" width="30%"/>

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const App());
}

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  HomePage({super.key});

  final _textController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Spell check bug'),
      ),
      body: TextField(
        controller: _textController,
        spellCheckConfiguration: const SpellCheckConfiguration(),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

Unfortunately the verbose logs were too long for GitHub, but you can still see the error it gave in the Output section above.

flutter analyze:

```
Analyzing spell_check_bug...                                            
No issues found! (ran in 1.0s)
```

flutter doctor -v:

```
[✓] Flutter (Channel beta, 3.7.0-1.1.pre, on Arch Linux 6.0.12-arch1-1, locale en_GB.UTF-8)
    • Flutter version 3.7.0-1.1.pre on channel beta at /home/sasha/programs/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e599f02c7a (8 days ago), 2022-12-08 01:07:24 +0530
    • Engine revision 4971c5b2e7
    • Dart version 2.19.0 (build 2.19.0-444.1.beta)
    • DevTools version 2.20.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/sasha/programs/Android/Sdk/
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 14.0.6
    • cmake version 3.25.1
    • ninja version 1.11.1
    • pkg-config version 1.8.0

[✓] Android Studio (version 2021.3)
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Connected device (2 available)
    • FP4 (mobile)    • d35ebb80 • android-arm64 • Android 11 (API 30)
    • Linux (desktop) • linux    • linux-x64     • Arch Linux 6.0.12-arch1-1

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Thanks for the report! This is a known issue being tracked at #34688. There is a filed PR for Android recently at https://github.com/flutter/flutter/pull/114460.
Closing this issue as a duplicate, so please follow up on it for further updates.

**Comment by Voklen** [comment]
Sorry, just checked and it works in an Pixel a3 API 33 emulator, so I think it's something with my phone.
Model: Fairphone 4
Android version: 11
Kernel version: 4.19.157-perf+
Build number: FP4.FP4G.A.170.20220920

**Comment by camsim99** [comment]
@huycozy This is a reported bug with the parts of spell check that were already landed. This is a real bug and I will look into it!

**Comment by camsim99** [comment]
@Voklen I wasn't able to reproduce this. I ran your app on a Pixel 3A, and it worked as expected. What device are you using?

**Comment by Voklen** [comment]
Model: Fairphone 4
Android version: 11
Build number: FP4.FP4G.A.170.20220920

**Comment by camsim99** [comment]
@Voklen What spell checker is your device using? You should be able to find out in the settings.

**Comment by tneotia** [comment]
**Samsung Spell Checker**
I'm also having issues with spell check - in my case it seems to be underlining all words regardless of whether it is spelled correctly or not. Tapping the word does not provide a list of suggestions to correct the spelling.

**Gboard Spell Checker**
Spell check does not provide any underlining or suggestions for incorrectly spelled words.

Device: Galaxy S10+
Spell checker: Tried with both Samsung Keyboard & Gboard spell checkers.
Language: English/US

**Comment by camsim99** [comment]
@tneotia The Samsung problem is known, so I just filed a bug for it https://github.com/flutter/flutter/issues/119534. I tried to run your app, but ran into some issues I think with my environment, but I haven't had issues with Gboard thus far. Is Gboard configured in any way on your device? For example, is personalization turned on? (As discussed [here](https://support.google.com/gboard/answer/12373137?hl=en))

Also, if you do get a full crash, could you file a separate issue so we can take a look? Thanks!

**Comment by tneotia** [comment]
@camsim99 all the options in the advanced menu of the gboard settings are turned on. I don't recall really messing with anything, but I've also had the phone for 3+ years so its possible I did a while ago.

As for running the app, it wouldn't do much good as it requires a login, but I can do any tests if necessary!

I tested Gboard, and yes it does appear to work actually! I guess some words that I think should be mispelled are not marked as misspelled, but it appears Google Messages has this same behavior.
